### PR TITLE
Enable usage as a Widget

### PR DESCRIPTION
**Pull request by Oksydan**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Implementing widget interface
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/18817
| How to test?  | Add productcomments module to e.g. displayReassurance or add it to product.tpl with {widget name="productcomments"}. Check product page and quickview.

### CODE REVIEW

**Comment by Oksydan** [comment]
@Progi1984 

IMO we don't need it anymore since widget interface is available.
Hook `displayProductAdditionalInfo ` is registered during installation and its output will be renderWidget.

**Comment by Progi1984** [comment]
@Oksydan In this case, the `hookDisplayProductListReviews` could have the same refacto ?

**Comment by Oksydan** [comment]
Well `hookDisplayProductListReviews ` is custom hook so is it really needed?

**Comment by Oksydan** [comment]
@Progi1984 

I changed that hook `hookDisplayProductListReviews` will be rendered by renderWidget method.
To display reviews on product list use hook `hookDisplayProductListReviews` or `{widget name='productcomments' product=$product type="product_list"}` (param type and product is required).

**Comment by Oksydan** [comment]
@matks @Progi1984 
Why this PR got closed?

**Comment by matks** [comment]
> @matks @Progi1984
> Why this PR got closed?

Not intended. It seems it was an automatic move from github 🤔  (my name is displayed but I did not hit any button 😄  )

File: productcomments.php [hunk 1]
```diff
@@ -922,65 +923,54 @@ private function renderProductCommentModal($product)
...
+            $id_product = $product['id_product'];
+            $variables = $this->getWidgetVariables($hookName, array('id_product' => $id_product));
+
+            $variables = array_merge($variables, array(
+                'product' => $product,
+                'product_comment_grade_url' => $this->context->link->getModuleLink('productcomments', 'CommentGrade')
+            ));
+
+            $file_path = 'module:productcomments/views/templates/hook/product-list-reviews.tpl';
+
```
Comment (sowbiba) [inline productcomments.php:957]: Please remove blank lines in the beginning and end of if/elseif

File: productcomments.php [hunk 2]
```diff
@@ -922,65 +923,54 @@ private function renderProductCommentModal($product)
...
+        } elseif ($this->context->controller instanceof ProductControllerCore) {
 
-        if ('quickview' === Tools::getValue('action')) {
-            return $this->context->smarty->fetch('module:productcomments/views/templates/hook/product-additional-info-quickview.tpl');
+            $id_product = Tools::getValue('id_product');
```
Comment (sowbiba) [inline productcomments.php:960]: You can use camelcase for variables names

File: productcomments.php [hunk 3]
```diff
@@ -922,65 +923,54 @@ private function renderProductCommentModal($product)
...
+            $variables = $this->getWidgetVariables($hookName, array('id_product' => $id_product));
+
+            $file_path = 'quickview' === Tools::getValue('action') ? 'module:productcomments/views/templates/hook/product-additional-info-quickview.tpl' : 'module:productcomments/views/templates/hook/product-additional-info.tpl';
+    
+        }
+
+        if (empty($variables)) {
```
Comment (sowbiba) [inline productcomments.php:967]: I understand that if we dont get into the IF or ELSEIF $variables is undefined. Can you initialize it with $variables = [] before the conditions ?

File: productcomments.php [hunk 4]
```diff
@@ -27,13 +27,14 @@
...
+use PrestaShop\PrestaShop\Core\Module\WidgetInterface;
 use PrestaShop\PrestaShop\Adapter\Presenter\Product\ProductLazyArray;
 use PrestaShop\Module\ProductComment\Repository\ProductCommentCriterionRepository;
 use PrestaShop\Module\ProductComment\Repository\ProductCommentRepository;
 use PrestaShop\Module\ProductComment\Addons\CategoryFetcher;
 
-class ProductComments extends Module
-{
+class ProductComments extends Module implements WidgetInterface
+{
```
Comment (PierreRambaud) [inline productcomments.php:37]:
```suggestion
{
```

**Review by matthieu-rolland — COMMENTED** [review]
lgtm ! there's just sowbiba's feedback to be done before we can aprove

**Comment by Oksydan** [comment]
@sowbiba 

Thank you for your review. The changes has been applied.

**Review by matthieu-rolland — APPROVED** [review]

**Comment by Oksydan** [comment]
@kpodemski done.

**Comment by sarahdib** [comment]
Hello @Oksydan 

I try to add {widget name="productcomments"} to product.tpl  but the comment block isn't displayed.

**Comment by Oksydan** [comment]
@sarahdib 

Hi.
Are you sure are adding it properly?
For templates/catalog/product.tpl {widget name="productcomments"}.
For templates/catalog/_partials/miniatures/product.tpl widget have to be {widget name='productcomments' product=$product type="product_list"}.

Example on templates/catalog/product.tpl:
Zrzut ekranu 2020-09-19 o 09 27 24

Result on product page:
Zrzut ekranu 2020-09-19 o 09 27 33

**Comment by sarahdib** [comment]
@Oksydan Thank you for the clarification, Now it's ok :)

**Comment by matks** [comment]
Thank you @Oksydan @kpodemski and @sarahdib 😄

**Pull request merged**